src/Types/Encoder.cs

```diff
@@ -1483,14 +1511,40 @@ static string ReadString(ByteBuffer buffer, byte formatCode, byte code8, byte co
             {
                 throw InvalidFormatCodeException(formatCode, buffer.Offset);
             }
+                       
+            buffer.ValidateRead(count);
 
-            buffer.Validate(false, count);
+#if NETMF
             string value = new string(Encoding.UTF8.GetChars(buffer.Buffer, buffer.Offset, count));
+#else
+            string value;
+            if (count <= MaxBytesForStackalloc)
+            {
+                int charCount = Encoding.UTF8.GetCharCount(buffer.Buffer, buffer.Offset, count);
+                if (charCount == 0)
+                {
+                    value = "";
+                }
+                else
+                {
+                    char* chars = stackalloc char[charCount];
```

Comment:
The only benefit of the stackalloc version was to avoid the allocation of the temporary char[] on the heap. But I tested the GetString() overload and that seems to be a much better choice indeed. Initially I thought that this overload allocates a char[] internally but I must have mixed that up with some other overload.

```
                Method | StringLength |        Mean |     Error |    StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
---------------------- |------------- |------------:|----------:|----------:|------------:|------------:|------------:|--------------------:|
        ReadString_Old |            1 |    59.08 ns | 0.5208 ns | 0.4872 ns |      0.0203 |           - |           - |                32 B |
 ReadString_Stackalloc |            1 |    66.10 ns | 0.5256 ns | 0.4389 ns |      0.0101 |           - |           - |                16 B |
  ReadString_GetString |            1 |    39.73 ns | 0.1655 ns | 0.1467 ns |      0.0101 |           - |           - |                16 B |
        ReadString_Old |          128 |   260.15 ns | 1.8914 ns | 1.4767 ns |      0.3428 |           - |           - |               540 B |
 ReadString_Stackalloc |          128 |   274.61 ns | 1.9875 ns | 1.8591 ns |      0.1726 |           - |           - |               272 B |
  ReadString_GetString |          128 |   161.47 ns | 0.8339 ns | 0.7801 ns |      0.1729 |           - |           - |               272 B |
        ReadString_Old |         1024 | 1,561.75 ns | 7.1132 ns | 6.3057 ns |      2.6207 |           - |           - |              4128 B |
 ReadString_Stackalloc |         1024 | 1,556.94 ns | 9.4310 ns | 8.8218 ns |      2.6207 |           - |           - |              4128 B |
  ReadString_GetString |         1024 |   974.78 ns | 7.0935 ns | 6.6353 ns |      1.3103 |           - |           - |              2064 B |